src/java/sliding_window/P3_alt_sol_2.java

```diff
@@ -0,0 +1,50 @@
+import java.util.HashSet;
+import java.util.Set;
+
+public class P3_alt_sol_2 {
+    
+     // @lc app=leetcode id=3 lang=java
+    // @lc code=start
+    public class Solution {
+         public int lengthOfLongestSubstring_sol_2(String s) {
+        
+        if(s == null ) {
+            return 0;
+        }
+
+        int len = s.length();
+
+        if(len ==1){
+            return 1;
+        }
+
+        Set<Character> uniChars = new HashSet<>();
+
+        int left=0 , right  =0;
+        int maxLen = 0;
+        while (right <len) {
+            if(!uniChars.contains(s.charAt(right))){
+                 uniChars.add(s.charAt(right));
+                right ++;
+               
+            } else{
+                 maxLen = Math.max(right-left , maxLen);
```

Comment:
I think you need to keep removing chars from the set until it no loger contains the right char.
the repeating char could be inside the window right..?